### DISCUSSION TITLE
flake8/black/isort → ruff?

### DISCUSSION THREAD

**Discussion by DimitriPapadopoulos** [nipreps/GOVERNANCE#45] [Development]
Would you agree to moving some repositories from `flake8`/`black`/`isort` to `ruff`/`ruff format`?

See for example https://github.com/pypa/setuptools/pull/4125#issuecomment-1822842090 for context:
> I’m pleased to see the use of Ruff over pyupgrade. Id also be interested to see Ruff replace black and implement other common formatting/linking like import sorts.

**Comment by effigies** [comment]
In a recent developer call, we accepted https://github.com/nipreps/smriprep/pull/397 and the plan is to open a similar PR on each repository after a minor release.

Current repositories that could accept PRs include:

* niworkflows
* nireports
* sdcflows

For fMRIPrep we will wait on until the final release of 23.2.0, but then I do intend to do that.

Nice link to setuptools discussion! I've recently been looking at jaraco's skeleton approach (noticed updates in importlib-resources), and we may go for something similar in nipreps. We need someone to scope it out and determine how much effort it would be to transition and what the exact maintenance steps would be.

For now, updating pre-commit, pyproject and CI seems like a good approach. At some point we'll want to standardize CI...

cc @oesteban @mnoergaard to make sure they're aware of this discussion.

**Comment by oesteban** [comment]
Tagging @esavary here. I updated her today about the skeleton idea and she will prepare some materials to ensure an informed discussion on the next TechMon meeting (Jan 16)
